### PR TITLE
chore: Add more info on dependency management

### PR DESCRIPTION
**Pull request by bebraw**

### CODE REVIEW

File: manuscript/code-quality/05-dependencies.md [hunk 1]
```diff
@@ -1,22 +1,33 @@
-# Dependencies
+# Dependency Management
+
+While writing projects, dependency management tends to be a common issue. Although lockfiles can help, you still have to keep up with the development as packages you depend upon evolve.
```
Comment (sapegin) [inline manuscript/code-quality/05-dependencies.md]: How lock files can help?

File: manuscript/code-quality/05-dependencies.md [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
+
+Sometimes this means you have to rewrite code. Bigger projects may provide **codemods** that can perform the required changes for you. They can also rely on deprecation patterns that tell you to rewrite certain pieces code before deprecating the functionality entirely.
```
Comment (sapegin) [inline manuscript/code-quality/05-dependencies.md]: “before deprecating” → ”before removing”.

File: manuscript/code-quality/05-dependencies.md [hunk 3]
```diff
@@ -1,22 +1,33 @@
...
 
-Several tools can help you with updates:
+```bash
+Package                 Current  Wanted  Latest  Location
+catalog                   3.0.0   3.1.2   3.1.2  react-component-boilerplate
+eslint-config-airbnb     15.1.0  15.1.0  16.0.0  react-component-boilerplate
+eslint-plugin-jsx-a11y    5.1.1   5.1.1   6.0.2  react-component-boilerplate
+```
+
+The output tells that in this case running [npm update](https://docs.npmjs.com/cli/update) could update `catalog` to `3.1.2` given it follows SemVer. The other dependencies would require either a manual change to *package.json* or using a specific tool:
```
Comment (sapegin) [inline manuscript/code-quality/05-dependencies.md]: “given it follows SemVer” — even if it doesn‘t, you may only guess and pray that they have a change log ;-)

File: manuscript/code-quality/05-dependencies.md (hunk 1 above)
Comment (bebraw) [reply manuscript/code-quality/05-dependencies.md]: It feels too specific idea. It's covered in detail earlier in the book. I'll add a tip with a link.

Comment (sapegin) [reply manuscript/code-quality/05-dependencies.md]: Then we should just remove this piece ;-)

Comment (bebraw) [reply manuscript/code-quality/05-dependencies.md]: Yup, I rewrote the intro. It doesn't feel great yet but it's better.

**Pull request merged**